src/main/java/com/session/service/client/SessionsClientImpl.java

```diff
@@ -48,7 +48,7 @@ public SessionCreated createNewSession(String userEmail) throws SessionClientExc
     private SessionCreated postSession(String userEmail) {
         try {
             String jsonStr = http.toJson(new CreateNewSession(userEmail));
-            HttpPost httpPost = http.createHttpPost(host, "/session", jsonStr);
+            HttpPost httpPost = http.createHttpPost(host, "/sessions", jsonStr);
```

Comment:
Add validation on the email - its a required field and cannot be null or empty. Checking and erroring now is more efficient than sending a request we know will fail.